src/nspawn/nspawn-network.c

```diff
@@ -84,96 +84,95 @@ static int generate_mac(
+                r = sd_netlink_message_open_container(m, VETH_INFO_PEER);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to open netlink container: %m");
 
-        r = sd_netlink_message_append_ether_addr(m, IFLA_ADDRESS, &mac_container);
-        if (r < 0)
-                return log_error_errno(r, "Failed to add netlink MAC address: %m");
+                r = sd_netlink_message_append_string(m, IFLA_IFNAME, *b);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to add netlink interface name: %m");
 
-        r = sd_netlink_message_append_u32(m, IFLA_NET_NS_PID, pid);
-        if (r < 0)
-                return log_error_errno(r, "Failed to add netlink namespace field: %m");
+                r = sd_netlink_message_append_ether_addr(m, IFLA_ADDRESS, &mac_container);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to add netlink MAC address: %m");
 
-        r = sd_netlink_message_close_container(m);
-        if (r < 0)
-                return log_error_errno(r, "Failed to close netlink container: %m");
+                r = sd_netlink_message_append_u32(m, IFLA_NET_NS_PID, pid);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to add netlink namespace field: %m");
 
-        r = sd_netlink_message_close_container(m);
-        if (r < 0)
-                return log_error_errno(r, "Failed to close netlink container: %m");
+                r = sd_netlink_message_close_container(m);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to close netlink container: %m");
 
-        r = sd_netlink_message_close_container(m);
-        if (r < 0)
-                return log_error_errno(r, "Failed to close netlink container: %m");
+                r = sd_netlink_message_close_container(m);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to close netlink container: %m");
 
-        r = sd_netlink_call(rtnl, m, 0, NULL);
-        if (r < 0)
-                return log_error_errno(r, "Failed to add new veth interfaces (host0, %s): %m", iface_name);
+                r = sd_netlink_message_close_container(m);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to close netlink container: %m");
 
-        i = (int) if_nametoindex(iface_name);
-        if (i <= 0)
-                return log_error_errno(errno, "Failed to resolve interface %s: %m", iface_name);
+                r = sd_netlink_call(rtnl, m, 0, NULL);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to add new veth interfaces (%s, %s): %m", *a, *b);
+
+                i = (int) if_nametoindex(*a);
+                if (i <= 0)
+                        return log_error_errno(errno, "Failed to resolve interface %s: %m", *a);
+        }
 
         return i;
```

Comment:
I'm not so sure about this.
Previously this would return *the* network interface link corresponding to the virtual machine.
Now we can create multiple, but we only register the *last* network interface with machined.